notes/books/designing-data-intensive-apps.md

```diff
@@ -565,3 +565,48 @@ synchronizing reads/writes on it, special care must be taken about which guarant
 provides. In a non-linearizable database you would need to design the solution differently or
 implement some domain specific compare-and-set strategy, like receiving a hash of the image on the
 message and if the hash doesn't match fail (with some retry strategy).
+
+## CAP Theorem
+
+There is some interesting remarks on the limitations of the CAP theorem. It was useful as a tool
+to start discussions a decade ago but the author defends that it is an outdated idea that
+is not very useful when discussing trade-offs in distributed systems. Some limitations include:
+
+1 - Easilly misunderstood as a binary trade-off, like you lose availabiliy to get consistency and vice
+versa. Trade-offs in distributed systems are much more nuanced and complex than this.
+
+2 - Introduces partitioning as a variable on the equation, which on distributed systems doesn't apply since
+it would imply single node systems (to avoid partitioning). So in the end most discussions end up being
+about consistency VS availability.
+
+3 - Even thinking on terms of consistency VS availability doesn't provide a complete model.
+It ignores another major architectural property that must be taken in consideration: performance.
+
+One of the points that made me think the most is performance because there is an concrete example
+that makes it very obvious the limitations of CAP when thinking about consistency models, the memory
+models of multi-core SMP CPUs.
```

Comment:
That grammar here is not quite right. It should be:

```suggestion
One of the points that made me think the most is performance because there is a concrete example
that makes the limitations of CAP when thinking about consistency models very obvious, the memory
models of multi-core SMP CPUs.
```

Or maybe this:

> One of the points that made me think the most is performance because there is a concrete example
that really shows the limitations of CAP when thinking about consistency models, the memory
models of multi-core SMP CPUs.